src/main/java/sdk/sample/common/CommonSdk.java

```diff
@@ -0,0 +1,321 @@
+     * @param intervalInSec Time in second that the function will poll to see if the resource has been deleted
+     * @param retries Number of times polling will be performed
+     * @param clazz Valid class types: NetAppAccountInner, CapacityPoolInner, VolumeInner, SnapshotInner
+     */
+    public static <T> void waitForNoANFResource(NetAppManagementClient anfClient, String resourceId, int intervalInSec, int retries, Class<T> clazz)
+    {
+        for (int i = 0; i < retries; i++)
+        {
+            Utils.threadSleep(intervalInSec * 1000);
+
+            try
+            {
+                switch (clazz.getSimpleName())
+                {
+                    case "NetAppAccountInner":
+                        NetAppAccountInner account = anfClient.getAccounts().getByResourceGroup(ResourceUriUtils.getResourceGroup(resourceId),
+                                ResourceUriUtils.getAnfAccount(resourceId));
+                        if (account == null)
+                            return;
+
+                        continue;
+
+                    case "CapacityPoolInner":
+                        CapacityPoolInner pool = anfClient.getPools().get(ResourceUriUtils.getResourceGroup(resourceId),
+                                ResourceUriUtils.getAnfAccount(resourceId),
+                                ResourceUriUtils.getAnfCapacityPool(resourceId));
+                        if (pool == null)
+                            return;
+
+                        continue;
+
+                    case "VolumeInner":
+                        VolumeInner volume = anfClient.getVolumes().get(ResourceUriUtils.getResourceGroup(resourceId),
+                                ResourceUriUtils.getAnfAccount(resourceId),
+                                ResourceUriUtils.getAnfCapacityPool(resourceId),
+                                ResourceUriUtils.getAnfVolume(resourceId));
+                        if (volume == null)
+                            return;
+
+                        continue;
+
+                    case "SnapshotInner":
+                        SnapshotInner snapshot = anfClient.getSnapshots().get(ResourceUriUtils.getResourceGroup(resourceId),
+                                ResourceUriUtils.getAnfAccount(resourceId),
+                                ResourceUriUtils.getAnfCapacityPool(resourceId),
+                                ResourceUriUtils.getAnfVolume(resourceId),
+                                ResourceUriUtils.getAnfSnapshot(resourceId));
+                        if (snapshot == null)
+                            return;
+                }
+            }
+            catch (Exception e)
+            {
+                Utils.writeWarningMessage(e.getMessage());
+                break;
+            }
+        }
+    }
+
+    public static void waitForVolumeSuccess(NetAppManagementClient anfClient, String resourceGroupName, String accountName, String poolName, String volumeName)
```

Comment:
I would say to be safer, lets get the old function back "waitForAnfResource" where it's more generic and check the success status for any ANF resource. and this function needs to be called after the creation of each resource.